src/Language/PureScript/CodeGen/JS.hs

```diff
@@ -475,7 +461,7 @@ moduleToJs (Module _ coms mn _ imps exps reExps foreigns decls) foreignInclude =
       done'' <- go done' bs'
       js <- binderToJs propVar done'' binder
       return (AST.VariableIntroduction Nothing propVar (Just (accessorString prop (AST.Var Nothing varName))) : js)
-  literalToBinderJS varName done (ArrayLiteral bs) = do
+  literalToBinderJS _ varName done (ArrayLiteral bs) = do
```

Comment:
Looks like this is what causes the array binders in `case _ of` syntax to no longer have a source span.

Reply:
Same, that's not what causes this.